contracts/savings/SavingsContract.sol

```diff
@@ -446,11 +479,12 @@ contract SavingsContract is ISavingsContractV3, Initializable, InitializableToke
         }
 
         // Burn required credits from the sender FIRST
-        _burn(msg.sender, credits_);
+        _burn(owner, credits_);
 
         // Optionally, transfer tokens from here to sender
         if (_transferUnderlying) {
-            require(underlying.transfer(msg.sender, underlying_), "Must send tokens");
+            require(underlying.transfer(receiver, underlying_), "Must send tokens");
```

Comment:
Newer mStable contracts use `safeTransfer` which comes from @openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol.  These need to add `using SafeERC20 for IERC20;` at the start of the contract.